### PR TITLE
feat: add @capacitor-mlkit/face-detection package

### PR DESCRIPTION
**Pull request by trancee**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] The changes have been tested successfully.
- [ ] A changeset has been created (`npm run changeset`).

### CODE REVIEW

File: packages/face-detection/src/definitions.ts [hunk 1]
```diff
@@ -12,18 +12,29 @@ export interface FaceDetectionPlugin {
...
+       */
+      path: string;
+    }
+  | {
+      /**
+       * Represents an image object (Base64 encoded).
+       *
+       * @since 5.1.0
+       */
+      image: string;
```
Comment (robingenz) [inline packages/face-detection/src/definitions.ts:30]: I would like to remove this option. This feature can lead to OOM exceptions. You should always try to use a native file path. If you really only have the image available in Base64 format in the webview, then you can still write it to the filesystem using the Capacitor Filesystem plugin (or the Capacitor Blob Writer plugin). But this prevents issues with this plugin regarding OOM.

File: packages/face-detection/src/definitions.ts [hunk 2]
```diff
@@ -12,18 +12,29 @@ export interface FaceDetectionPlugin {
...
-   * @since 5.1.0
+   * The options for the face detector.
    */
-  path: string;
-  /**
-   * The classification mode for face detection.
-   */
-  classificationMode: ClassificationMode;
-}
+  options?: FaceDetectorOptions;
```
Comment (robingenz) [inline packages/face-detection/src/definitions.ts:36]: Please remove the `FaceDetectorOptions` interface and just add the properties to `ProcessImageOptions` for now.

**Review by robingenz — CHANGES_REQUESTED** [review]

**Comment by trancee** [comment]
As for the enums: I am not quite sure what the naming convention here is for those. Currently, I am using the same as in Google's documentation for Android (all UPPERCASE), but for iOS it's all lowercase. 
```
export enum PerformanceMode {
  FAST = 1,
  ACCURATE = 2,
}
```
What is your opinion?

**Comment by robingenz** [comment]
> As for the enums: I am not quite sure what the naming convention here is for those. Currently, I am using the same as in Google's documentation for Android (all UPPERCASE), but for iOS it's all lowercase.
> 
> ```
> export enum PerformanceMode {
>   FAST = 1,
>   ACCURATE = 2,
> }
> ```
> 
> What is your opinion?

@trancee As far as I know, in TypeScript you typically use PascalCase. This convention is also used for the official Capacitor plugins ([example](https://github.com/ionic-team/capacitor-plugins/blob/922d55d16f7d00b5880fde902977befcf3c92fc5/filesystem/src/definitions.ts#L24)).

```ts
export enum PerformanceMode {
  Fast = 1,
  Accurate = 2,
}
```

File: package-lock.json
Comment (robingenz) [inline package-lock.json:1]: Please also revert the changes in this file.

File: packages/face-detection/src/definitions.ts [hunk 3]
```diff
@@ -34,23 +78,438 @@ export interface ProcessImageResult {
...
+  points: PointF[];
+}
+
+/**
+ * PointF holds two float coordinates
+ *
+ * @since 5.1.0
+ * @see https://developer.android.com/reference/android/graphics/PointF.html
+ */
+export interface PointF {
```
Comment (robingenz) [inline packages/face-detection/src/definitions.ts:342]:
```suggestion
export interface Point {
```

Let's rename this to `Point`. `PointF` might confuse many and there is no other interface named `Point` anyway.

**Review by robingenz — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (robingenz) [inline package-lock.json:1]: There are still changes in this file. Try the following:

```bash
git checkout feat/face-detection package-lock.json
```

**Review by robingenz — CHANGES_REQUESTED** [review]

**Comment by robingenz** [comment]
Please also run `npm run docgen && npm run fmt` again.

**Review by robingenz — APPROVED** [review]

**Comment by robingenz** [comment]
The build failed only because of iOS, so I merge it.

**Comment by trancee** [comment]
> The build failed only because of iOS, so I merge it.

I will work on the iOS part now, should I create new PR for this?

**Comment by robingenz** [comment]
> > The build failed only because of iOS, so I merge it.
> 
> I will work on the iOS part now, should I create new PR for this?

I just saw your comment now. Yes, we do the same again.
